test/functional/wallet_listreceivedby.py

```diff
@@ -167,5 +169,117 @@ def run_test(self):
+
+        self.log.info("getreceivedbylabel returns reward with defaults")
+        balance = self.nodes[0].getreceivedbylabel("label")
+        assert_equal(balance, reward)
+
+        self.log.info("listreceivedbyaddress includes address with defaults")
+        assert_array_result(self.nodes[0].listreceivedbyaddress(),
+                            {"address": address},
+                            {"address": address, "amount": reward})
+
+        self.log.info("listreceivedbylabel includes label with defaults")
+        assert_array_result(self.nodes[0].listreceivedbylabel(),
+                            {"label": label},
+                            {"label": label, "amount": reward})
+
+        self.log.info("Orphan block that paid to address")
+        self.nodes[0].invalidateblock(hash)
+
+        self.log.info("getreceivedbyaddress does not include orphan when minconf is 0 when including immature")
+        balance = self.nodes[0].getreceivedbyaddress(address=address, minconf=0, include_immature=True)
+        assert_equal(balance, 0)
+
+        self.log.info("getreceivedbylabel does not include orphan when minconf is 0 when including immature")
+        balance = self.nodes[0].getreceivedbylabel(label="label", minconf=0, include_immature=True)
+        assert_equal(balance, 0)
+
+        self.log.info("listreceivedbyaddress does not include orphan when minconf is 0 when including immature")
+        assert_array_result(self.nodes[0].listreceivedbyaddress(minconf=0, include_immature=True),
+                            {"address": address},
+                            {}, True)
+
+        self.log.info("listreceivedbylabel does not include orphan when minconf is 0 when including immature")
+        assert_array_result(self.nodes[0].listreceivedbylabel(minconf=0, include_immature=True),
+                            {"label": label},
+                            {}, True)
+
+
+        self.log.info("Generate 101 blocks to node with deprecated filter_coinbase")
+        address = self.nodes[1].getnewaddress(label)
+        self.nodes[1].generatetoaddress(101, address)
+
+        self.log.info("getreceivedbyaddress returns nothing when filtering coinbase")
+        balance = self.nodes[1].getreceivedbyaddress(address)
+        assert_equal(balance, 0)
+
+        self.log.info("getreceivedbylabel returns nothing when filtering coinbase")
+        balance = self.nodes[1].getreceivedbylabel("label")
+        assert_equal(balance, 0)
+
+        self.log.info("listreceivedbyaddress does not include address when filtering coinbase")
+        assert_array_result(self.nodes[1].listreceivedbyaddress(),
+                            {"address": address},
+                            {}, True)
+
+        self.log.info("listreceivedbylabel does not include label when filtering coinbase")
+        assert_array_result(self.nodes[1].listreceivedbylabel(),
+                            {"label": label},
+                            {}, True)
+
```

Comment:
See the above comment about `include_immature` and `"-deprecatedrpc=filter_coinbase"`. Perhaps add a test for that combination?